src/univariate.jl

```diff
@@ -66,23 +70,34 @@ function kde_range(boundary::(@compat Tuple{Real,Real}), npoints::Int)
     lo:step:hi
 end
 
+immutable UniformWeights
+    w
+
+    UniformWeights(n) = new(1/n)
+end
+
+Base.sum(x::UniformWeights) = 1.
+Base.getindex(x::UniformWeights, i) = x.w
+
+typealias Weights Union{UniformWeights, RealVector, WeightVec}
+
+
```

Comment:
You could have the length be a part of the type parameter so that its slightly more efficient.
```julia
type UniformWeights{N} end
UniformWeights(N) = UniformWeights{N}()

Base.sum(x::UniformWeights) = 1.0
Base.getindex{N}(x::UniformWeights{N}, i) = 1/N
Base.length{N}(x::UniformWeights{N}) = N
values{N}(x::UniformWeights{N}) = fill(x[1], N)
eltype(x::UniformWeights) = eltype(x[1])
isempty(x::UniformWeights) = false
```